### DISCUSSION TITLE
Compilation error for FTPD

### DISCUSSION THREAD

**Discussion by mdnnetworksolutionsllc** [MarcJHuber/event-driven-servers#113] [General]
Hello,

I'm having an issue with making the TACACS_NG and its giving me an error on this:

make[2]: Entering directory '/home/mdnadmin/tac_plugng/event-driven-servers/build/linux-6.8.0-45-generic-x86_64/ftpd'
gcc  -Wall -W -Wno-strict-prototypes -Wno-implicit-fallthrough  -DOSTYPE=linux -DOSLEVEL=0x06080000 -DOS="\"linux-6.8.0-45-generic-x86_64\"" -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DWITH_ARES -DWITH_DNS -DWITH_CRYPTO -DWITH_PCRE2 -DPCRE2_CODE_UNIT_WIDTH=8 -DWITH_CRYPT -DWITH_SCTP -DWITH_PTHREAD -DWITH_LDAP -DWITH_LBER -DWITH_EPOLL -DWITH_POLL -DWITH_SELECT -DWITH_IPC -I/home/mdnadmin/tac_plugng/event-driven-servers  -I/usr/include   -c -o h_pbsz.o /home/mdnadmin/tac_plugng/event-driven-servers/ftpd/h_pbsz.c
/home/mdnadmin/tac_plugng/event-driven-servers/ftpd/h_pbsz.c: In function ‘h_pbsz’:
/home/mdnadmin/tac_plugng/event-driven-servers/ftpd/h_pbsz.c:21:13: error: ‘struct context’ has no member named ‘ssl_c’

mdnadmin@mdntacplus:~/tac_plugng/event-driven-servers$ ./configure --with-pcre2 --with-ldap --with-ares --with-ssl --with-tls
==============================================================================================================================================================================================================
Development files were not found for: LIB-CURL, LIB-FREERADIUS_CLIENT, LIB-PAM, LIB-RADCLI, LIB-ZLIB

Development files were found for: LIB-ARES, LIB-CRYPT, LIB-CRYPTO, LIB-LBER, LIB-LDAP, LIB-PCRE2, LIB-PTHREAD, LIB-SCTP

The file

    build/Makefile.inc.linux-6.8.0-45-generic-x86_64

was left unchanged. You may run "make" now. After that, you may wish to do a "make install". Alternatively, you'll find the compiled binaries (plus some ancillary scripts) in the

    build/linux-6.8.0-45-generic-x86_64/fakeroot/

directory structure. It's probably advisable to have a look there in any case, as you may or may not like the particular file system layout, and this will give you a chance to see it before installing.

Please direct support requests to the "Event-Driven Servers" Google Group at

    event-driven-servers@googlegroups.com
    https://groups.google.com/group/event-driven-servers

or file an issue at

    https://github.com/MarcJHuber/event-driven-servers

Support requests sent to the author's private email address may be silently ignored.

Can you please assist? Clearly Im missing a library.

Regards,

Jason

**Comment by MarcJHuber** [comment]
Hi Jason,

thanks, I'll have a look at that. This issue is most certainly caused by specifying both --with-ssl  (that's OpenSSL) and --with-tls (this one is for LibreTLS), I don't believe this happens with auto-detection.

Just running "./configure tac_plus-ng" is sufficient if you only care for tac_plus-ng.

Cheers,

Marc

**Comment by mdnnetworksolutionsllc** [comment]
Thanks for the supersonic response Marc, really appreciate that!!

That definitely did the trick. Removing both --with-ssl and --with-tls, doing a make clean and then a configure again without those two inclusion statements allowed make to run successfully without error.

Regards,
Jason
